aten/src/ATen/core/ivalue.h

```diff
@@ -518,16 +518,16 @@ struct CAFFE2_API IValue final {
   }
 
   // Detection Aliased tensors.
-  struct HashIValue {
+  struct HashAliasedIValue {
     size_t operator()(const IValue& val) const {
       if (val.isTensor()) {
-        return 0;
+        return reinterpret_cast<size_t>(val.toTensor().storage().unsafeGetStorageImpl());
       }
       return val.hash();
```

Comment:
Can you add a comment here? 
// if it is not a Tensor, then two mutable IValues alias each other only if they are the same pointer

Reply:
ok